spec/jobs/idv/profile_job_spec.rb

```diff
@@ -71,5 +70,19 @@
         expect(result.job_failed?).to eq(true)
       end
     end
+
+    it 'selects the proofer vendor based on the config' do
+      mock_proofer_job_agent(config: :proofing_vendor_profile, vendor: 'fancy_vendor')
+
+      Idv::ProfileJob.perform_now(
+        result_id: result_id,
+        vendor_params: vendor_params,
+        applicant_json: applicant_json
+      )
+      result = VendorValidatorResultStorage.new.load(result_id)
+
+      expect(Idv::Agent).to have_received(:new).with(hash_including(vendor: :fancy_vendor))
+      expect(result).to be_a(Idv::VendorResult)
+    end
```

Comment:
how about a test for `nil`/ empty env vars? I am thinking if, for some reasons, these env vars weren't uploaded to Citadel

Reply:
I think if these are nil, we want the app to crash. I'll write a spec in the `UpcaseEnvVars` service to make sure it raises when these are nil.

Reply:
Figaro has a built in feature to make sure keys are present: add the key you want to be required (`proofing_vendors`) in `config/initializers/figaro.rb`. However, if the key is present and set to an empty string, it won't raise an error. I think the best place for that would probably be the `ConfigValidator` class.